doc/architecture/decisions/0025-client-off-boarding.md

```diff
@@ -0,0 +1,38 @@
+# 25. Client off-boarding
+
+[Next >>](9999-end.md)
+
+Date: 2023-05-10
+
+## Status
+
+Proposed
+
+## Context
+
+When a client no longer wants to receive events from our system, we should have some way of off-boarding them. This
+should include stopping sending them events and marking them as deleted, we do not want to fully delete them from our DB
+until it is safe to do so with respect to data security.
+
+## Approach
+
+We should have 4 new endpoints, only accessible by the admin:
+
+1. Delete acquirer
+2. Delete acquirer subscription
+3. Delete supplier
+4. Delete supplier subscription
+
+Deleting the subscriptions is in case a supplier or acquirer would like to maintain another subscription, but remove
+this one. In these cases, we will:
+
+1. Mark the subscription as deleted, so that no new events get sent to/from it
+2. For acquirer subscriptions, mark all non-consumed acquirer events as consumed
+3. If there is a queue connected to the subscription, tag the queue as DeletedAt (with the date and time)
```

Comment:
Why tag and not delete the queue?

Reply:
My reasoning was that we would follow the same process as with the cognito clients, so that we don't lose any data.

I was thinking that if we tag them as deleted, then if we need to recover them due to an error we do not need to reissue keys